test/classifier/logistic_regressor/logistic_regressor_impl_test.dart

```diff
@@ -0,0 +1,172 @@
+    final negativeLabel = -1;
+    final positiveLabel = 1;
+    final dtype = DType.float32;
+
+    final regressor = LogisticRegressorImpl(
+      className,
+      linkFunctionMock,
+      fitIntercept,
+      interceptScale,
+      coefficients,
+      probabilityThreshold,
+      negativeLabel,
+      positiveLabel,
+      dtype,
+    );
+
+    final testFeatureMatrix = Matrix.fromList([
+      [20, 30, 40],
+      [20, 22, 11],
+      [90, 87, 52],
+      [12, 20, 21],
+      [33, 44, 55],
+    ]);
+
+    final featuresWithIntercept = Matrix.fromList([
+      [interceptScale, 20, 30, 40],
+      [interceptScale, 20, 22, 11],
+      [interceptScale, 90, 87, 52],
+      [interceptScale, 12, 20, 21],
+      [interceptScale, 33, 44, 55],
+    ]);
+
+    final mockedProbabilities = Matrix.fromList([
+      [0.8],
+      [0.5],
+      [0.6],
+      [0.7],
+      [0.3],
+    ]);
+
+    setUp(() {
+      when(linkFunctionMock.link(any)).thenReturn(mockedProbabilities);
+    });
+
+    tearDown(() {
+      reset(linkFunctionMock);
+    });
+
+    group('constructor', () {
+      test('should create the instance with `classNames` list of just one '
+          'element', () {
+        expect(regressor.classNames, equals([className]));
+      });
+
+      test('should throw an exception if no coefficients are provided', () {
+        final coefficients = Matrix.empty();
+        final actual = () => LogisticRegressorImpl(
+          className,
+          linkFunctionMock,
+          fitIntercept,
+          interceptScale,
+          coefficients,
+          probabilityThreshold,
+          negativeLabel,
+          positiveLabel,
+          dtype,
+        );
+
+        expect(actual, throwsException);
+      });
+
+      test('should throw an exception if coefficients provided for more than '
+          'one class', () {
+        final coefficients = Matrix.fromList([
+          [1, 2, 3],
+          [7, 5, 8],
+          [2, 0, 9],
+          [1, 3, 3],
+        ]);
+        final actual = () => LogisticRegressorImpl(
+          className,
+          linkFunctionMock,
+          fitIntercept,
+          interceptScale,
+          coefficients,
+          probabilityThreshold,
+          negativeLabel,
+          positiveLabel,
+          dtype,
+        );
+
+        expect(actual, throwsException);
+      });
+    });
+
+    group('predict method', () {
+      test('should throw an exception if no test features provided', () {
+        final testFeatures = DataFrame.fromMatrix(Matrix.empty());
+
+        expect(() => regressor.predict(testFeatures), throwsException);
+      });
+
+      test('should consider intercept term', () {
+        final testFeatures = DataFrame.fromMatrix(testFeatureMatrix);
+
+        regressor.predict(testFeatures);
+
+        verify(linkFunctionMock.link(featuresWithIntercept * coefficients))
+            .called(1);
+      });
+
+      test('should throw an error if provided too many features', () {
```

Comment:
the word `provided` should be placed in the end of the sentence, the same for below test-case